src/cmd/initialize.go

```diff
@@ -52,73 +52,84 @@ var initCmd = &cobra.Command{
 
-			// If the init-package doesn't exist in the executable directory, suggest trying to download
-			if utils.InvalidPath(config.DeployOptions.PackagePath) {
-
-				if config.CommonOptions.Confirm {
-					message.Fatalf(nil, "This command requires a zarf-init package, but one was not found on the local system.")
+				// If the init-package doesn't exist in the cache directory, suggest downloading it
+				if utils.InvalidPath(config.DeployOptions.PackagePath) {
+					if err := downloadInitPackage(initPackageName); err != nil {
+						message.Fatal(err, "Failed to download the init package")
+					}
 				}
+			}
+		}
+
+		// Run everything
+		packager.Deploy()
+	},
+}
 
-				// Parse the CLI version and extract its parts
-				initPackageVersion := strings.TrimLeft(config.CLIVersion, "v")
-				version, err := semver.StrictNewVersion(initPackageVersion)
+func downloadInitPackage(initPackageName string) error {
+	if config.CommonOptions.Confirm {
+		return fmt.Errorf("this command requires a zarf-init package, but one was not found on the local system")
+	}
 
-				if err != nil {
-					// If no CLI version exists (should only occur in dev or CI), try to get the latest release tag from Githhub
-					initPackageVersion, err = utils.GetLatestReleaseTag(config.GithubProject)
-					if err != nil {
-						message.Fatal(err, "No CLI version found and unable to get the latest release tag for the zarf cli.")
-					}
-				} else {
-					// If CLI version exists then get the latest init package for the matching major, minor and patch
-					initPackageVersion = fmt.Sprintf("v%d.%d.%d", version.Major(), version.Minor(), version.Patch())
-				}
+	// Parse the CLI version and extract its parts
+	initPackageVersion := strings.TrimLeft(config.CLIVersion, "v")
+	version, err := semver.StrictNewVersion(initPackageVersion)
 
-				var confirmDownload bool
-				url := fmt.Sprintf("https://github.com/%s/releases/download/%s/%s", config.GithubProject, initPackageVersion, initPackageName)
+	if err != nil {
+		// If no CLI version exists (should only occur in dev or CI), try to get the latest release tag from Githhub
+		initPackageVersion, err = utils.GetLatestReleaseTag(config.GithubProject)
+		if err != nil {
+			return fmt.Errorf("no CLI version found and unable to get the latest release tag for the zarf cli: %w", err)
+		}
+	} else {
+		// If CLI version exists then get the latest init package for the matching major, minor and patch
+		initPackageVersion = fmt.Sprintf("v%d.%d.%d", version.Major(), version.Minor(), version.Patch())
+	}
 
-				// Give the user the choice to download the init-package and note that this does require an internet connection
-				message.Question(fmt.Sprintf("It seems the init package could not be found locally, but can be downloaded from %s", url))
+	var confirmDownload bool
+	url := fmt.Sprintf("https://github.com/%s/releases/download/%s/%s", config.GithubProject, initPackageVersion, initPackageName)
```

Comment:
Now that the initPackageName contains the initPackageVersion this fails on anything other than a pure tagged release.  Previously (see line 103) it would chop off the -g0000000 commit piece when downloading which would still be nice to have for testing purposes and custom builds that do not change the agent.

Reply:
![image](https://user-images.githubusercontent.com/3977569/195243064-5a11de0b-aa64-4ca8-b5ba-4a95c43e07f0.png)

Reply:
But why would you want to d/l an old version of the init package. If you did that right now without this change it would break.  I'm not sure it's a valid use-case for a dev to download mismatched versions like that, except for upgrade testing, which should still be version matched anyway?

Reply:
^ it would break because master has breaking changes since the last release

Reply:
It would break currently, but was a feature in the past.  Either that stripping code should be removed and it should be a non-feature or it should be implemented fully.

Reply:
There is also code to get the latest release tag that should probably be dropped. The latest release tag will be out of sync with master in this case too.

Reply:
The likely *best* course of action may actually be to strip the version and compare it to the CLI version.  If mismatched, show a warning that you are in dev, and should manually create an init package.